### ISSUE TITLE
insert in transaction causing com.arangodb.ArangoDBException: Response: 500, Error: 4 - Builder value not yet sealed

### ISSUE THREAD

**Issue by cerveada** [arangodb/arangodb#12304]
## My Environment

* __ArangoDB Version__:       3.6.5
* __Storage Engine__:            RocksDB
* __Deployment Mode__:       Single Serve
* __Deployment Strategy__:  Manual Start
* __Configuration__:               
* __Infrastructure__:               own
* __Operating System__:        MacOS 10.15.4
* __Total RAM in your machine__:     32Gb
* __Disks in use__:        HDD
* __Used Package__:             brew

## Component, Query & Data

__Affected feature__:
insert in transaction via java driver

__Dataset__:
I was able to isolate just one document that is causing this issue, but haven't found anything special about it, maybe it's a bit bigger (500 kB) but that is not that huge. Most of the data in the document are in tree structure sveral levels deep.

__Size of your Dataset on disk__:
500 kB

## Steps to reproduce
js:
```
function (_params) {
  const _db = require('internal').db;
  _params.operation.forEach(o => _db.operation.insert(o, {silent:false}));
}
````
transaction call
```
db.transaction(js, classOf[Unit], txBuilder.options)
```
no data are saved and this exception is thrown:
```
java.util.concurrent.CompletionException: com.arangodb.ArangoDBException: Response: 500, Error: 4 - Builder value not yet sealed
	at java.util.concurrent.CompletableFuture.encodeRelay(CompletableFuture.java:326)
	at java.util.concurrent.CompletableFuture.completeRelay(CompletableFuture.java:338)
	at java.util.concurrent.CompletableFuture.uniRelay(CompletableFuture.java:911)
	at java.util.concurrent.CompletableFuture$UniRelay.tryFire(CompletableFuture.java:899)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1977)
	at com.arangodb.async.internal.velocystream.VstCommunicationAsync.lambda$execute$0(VstCommunicationAsync.java:66)
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760)
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:474)
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1962)
	at com.arangodb.async.internal.velocystream.VstConnectionAsync.lambda$write$0(VstConnectionAsync.java:48)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at com.arangodb.internal.velocystream.internal.MessageStore.consume(MessageStore.java:62)
	at com.arangodb.internal.velocystream.internal.ChunkStore.checkCompleteness(ChunkStore.java:65)
	at com.arangodb.internal.velocystream.internal.ChunkStore.checkCompleteness(ChunkStore.java:59)
	at com.arangodb.internal.velocystream.internal.VstConnection.lambda$open$0(VstConnection.java:151)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.arangodb.ArangoDBException: Response: 500, Error: 4 - Builder value not yet sealed
	... 15 common frames omitted
```

The code is tested, and we run it on many documents already without issues, until now.
There are no errors in arangodb log.

**Comment by cerveada** [comment]
I captured the data send from the driver and try to do the POST request directly and the issue is still there. This should be lot easier for you to replicate:

[json-query-operation-dummydata.txt](https://github.com/arangodb/arangodb/files/4987354/json-query-operation-dummydata.txt)

POST to /_api/transaction

**Comment by rashtao** [comment]
It can be reproduced executing:

```sh
curl -X POST "http://localhost:8529/_db/_system/_api/collection" -d '{"name": "operation"}'
curl -X POST "http://localhost:8529/_db/_system/_api/transaction" -d @json-query-operation-dummydata.txt
```

**Comment by maxkernbach** [comment]
Hi @cerveada,

Version 3.6.7, which includes a fix for the error message, has been released.

From the [Changelog](https://github.com/arangodb/arangodb/blob/edd53d086d812a21858d85152b900af86161e740/CHANGELOG#L219):

> * Fixed issue #12304: insert in transaction causing
>   com.arangodb.ArangoDBException: Response: 500, Error: 4 - Builder value not
>   yet sealed.
> 
>   This happened when too deeply-nested documents (more than 63 levels of
>   nesting) were inserted. While indefinite nesting is still not supported, the
>   error message has been corrected from the internal HTTP 500 error "Builder
>   value not yet sealed" to the correct HTTP 400 "Bad parameter".
